driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -51,28 +50,48 @@
      */
     <R extends Expression> ArrayExpression<R> map(Function<? super T, ? extends R> in);
 
+    IntegerExpression size();
+
+    BooleanExpression any(Function<T, BooleanExpression> map);
+
+    BooleanExpression all(Function<T, BooleanExpression> map);
+
+    NumberExpression sum(Function<T, NumberExpression> map);
+
+    NumberExpression multiply(Function<T, NumberExpression> map);
+
+    NumberExpression max(Function<T, NumberExpression> map, NumberExpression orElse);
+
+    NumberExpression min(Function<T, NumberExpression> map, NumberExpression orElse);
```

Comment:
I think we can still use `$min` for min, but then use a T for the return value. I like this.

I am not sure about implementing other operators beyond the "top-50", since otherwise there is no stopping.

Reply:
You were correct that implementing this using minN was best; I have added minN and maxN. I have named them `max` and `min` - perhaps the N should be included.

For changing min/max to be available everywhere: https://github.com/mongodb/mongo-java-driver/pull/1032/files#r1035214309